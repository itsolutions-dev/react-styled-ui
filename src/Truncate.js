import Text from './Text';

const Truncate = Text.extend`
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
`;

export default Truncate;
