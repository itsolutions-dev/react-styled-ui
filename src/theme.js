export default {
  radius: 8,
  spacing: {
    margin: 8,
    padding: 8,
    iconSize: 24,
    borderSize: 1,
    shadowSize: 1,
  },
  font: {
    size: 16,
    weight: 200,
    lineHeight: 18,
    family: 'Roboto, sans-serif',
  },
  h1: {
    size: 32,
    weight: 900,
    lineHeight: 64,
  },
  h2: {
    size: 24,
    weight: 700,
    lineHeight: 32,
  },
  h3: {
    size: 18,
    weight: 500,
    lineHeight: 24,
  },
  selected: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  hover: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  transitions: {
    in: undefined,
    out: undefined,
  },
  palette: {
    default: 'green',
    defaultText: 'white',
    defaultDisabled: undefined,
    defaultHover: undefined,
    defaultSelected: undefined,
    primary: 'red',
    primaryText: 'white',
    primaryDisabled: undefined,
    primaryHover: undefined,
    primarySelected: undefined,
    secondary: 'blue',
    secondaryText: 'white',
    secondaryDisabled: undefined,
    secondaryHover: undefined,
    secondarySelected: undefined,
    disabled: 'lightgray',
    shadow: 'black',
  },
};
