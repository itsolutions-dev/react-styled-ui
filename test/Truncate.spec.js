import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import theme from '../src/theme';
import Truncate from '../src/Truncate';

describe('Truncate', () => {
  test('should render a Truncate', () => {
    const component = ReactTestRenderer.create(<Truncate theme={theme} />);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have a p tag', () => {
    const component = ReactTestRenderer.create(<Truncate theme={theme} />).toJSON();
    expect(component.type).toEqual('p');
  });

  test('should have correct styles', () => {
    const component = ReactTestRenderer.create(<Truncate theme={theme} />);
    expect(component).toHaveStyleRule('overflow', 'hidden');
    expect(component).toHaveStyleRule('white-space', 'nowrap');
    expect(component).toHaveStyleRule('text-overflow', 'ellipsis');
  });
});
