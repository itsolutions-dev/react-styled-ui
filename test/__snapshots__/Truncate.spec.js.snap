// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Truncate should render a Truncate 1`] = `
.c0 {
  font-size: 16px;
  font-family: Roboto,sans-serif;
  font-weight: 200;
  line-height: 18px;
  color: #4c4c4c;
  margin: 0;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

<p
  className="c0"
/>
`;
